suite-native/device-manager/src/components/ConnectButton.tsx

```diff
@@ -0,0 +1,125 @@
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { TrezorDevice } from '@suite-common/suite-types';
+import {
+    selectHasRunningDiscovery,
+    selectInstacelessUnselectedDevices,
+    selectIsDeviceConnected,
+    selectSelectedDevice,
+} from '@suite-common/wallet-core';
+import { EventType, analytics } from '@suite-native/analytics';
+import { AnimatedBox, Button } from '@suite-native/atoms';
+import { FeatureFlag, useFeatureFlag } from '@suite-native/feature-flags';
+import { IconName } from '@suite-native/icons';
+import { Translation, TxKeyPath } from '@suite-native/intl';
+import {
+    AuthorizeDeviceStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+import { ANIMATION_DURATION } from './DeviceList';
+import { useDeviceManager } from '../hooks/useDeviceManager';
+
+type ConnectButtonProps = {
+    onSelectDevice: (device: TrezorDevice) => void;
+};
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    paddingTop: utils.spacings.sp32,
+    paddingHorizontal: utils.spacings.sp16,
+}));
+
+type NavigationProp = StackToStackCompositeNavigationProps<
+    RootStackParamList,
+    RootStackRoutes.AppTabs,
+    RootStackParamList
+>;
+
+export const ConnectButton = ({ onSelectDevice }: ConnectButtonProps) => {
+    const { applyStyle } = useNativeStyles();
+    const hasDiscovery = useSelector(selectHasRunningDiscovery);
+    const isDeviceConnected = useSelector(selectIsDeviceConnected);
+    const navigation = useNavigation<NavigationProp>();
+    const { setIsDeviceManagerVisible } = useDeviceManager();
+    const isBluetoothEnabled = useFeatureFlag(FeatureFlag.IsBluetoothEnabled);
+    const device = useSelector(selectSelectedDevice);
+    const notSelectedInstancelessDevices = useSelector(selectInstacelessUnselectedDevices);
+
+    const hasUnselectedDevices = notSelectedInstancelessDevices.length > 0;
+
+    const isOnlyBluetoothSupported = Platform.OS === 'ios' && isBluetoothEnabled;
+
+    const isConnectButtonVisible = !hasDiscovery && !isDeviceConnected;
+
+    const handleConnectDevice = () => {
+        if (device) {
+            onSelectDevice(device);
+        }
+        setIsDeviceManagerVisible(false);
+        if (isOnlyBluetoothSupported) {
+            // make sure the device manager is already hidden before navigating to prevent app freezing
+            // TODO: might be fixed by https://github.com/trezor/trezor-suite/issues/17968
+            setTimeout(
+                () =>
+                    navigation.navigate(RootStackRoutes.AuthorizeDeviceStack, {
+                        screen: AuthorizeDeviceStackRoutes.ConnectBluetoothDevice,
+                    }),
+                2 * ANIMATION_DURATION,
+            );
```

Comment:
I know you're not the author of this code, but is this necessary? I wonder how is this different from the else statement where we also redirect. IMO this should be removed.

Reply:
You mean the `timeout` no sure about that now it works just fine even without it. but other than that the reason is that this redirects to the `ConnectBluetoothDevice` and the else statement to `ConnectAndUnlockDevice`

Reply:
Yes, I understand the difference between these two statements is the target screen 😄 But I think behavior wise technically it should work same and if the timeout isn't required in one, it shouldn't be required in the other one (which you now say that it works even without the timeout?)

If so, I would like to remove it if it's unnecessary complexity please 🙏

Reply:
Oki, makes sense, I was worried that there could be some super edge case when a bluetooth device is connected and since mine devkit is flaky I did not want to take a chance, but it seems to be fine. here is a polished version https://github.com/trezor/trezor-suite/pull/20561/commits/51cd51e0863c4b927bb471de574974b3f4d3798c